### DISCUSSION TITLE
Setting USB Power and a few other newbie questions

### DISCUSSION THREAD

**Discussion by stevecolt** [earlephilhower/arduino-pico#1237] [Q&A]
I'm looking at using a Pi-Pico in a small USB-powered and connected instrument that involves LED lights (6x 20ma) and a small stepper motor that can take 300ma. So I really need to make sure that the USB once enumerated will deliver the maximum USB2.0  rated 500ma. (assumes the computer side port can be configured as well, but most seem to, as I've been running this hardware running with a Nano for a while, and only changed the micro.
I'm using The 2.73 Library and so far most things have worked really well, and seems far more compatible than the official embed solution which I wasted a week on before swapping.  I run this on VisualMicro under VisualStudio2019.
Whilst debugging my code I stumbled across a .json file that had an entry about 'set max USB power to 250ma', perhaps all I have to do is change this, but sadly I'm struggling to find it again. Was I dreaming? USBView also confirms max power is 250mA
I realise that 250mA is probably enough for most people and that it's probably set to limit power for the 3V3 to less than the 300ma for the regulator, but I'll be happy with <100ma for the 3v3 volts.
Thanks for any feedback.
Steve

**Comment by earlephilhower** [comment]
Yes, if you update the JSON being selected by Platform.IO, it will set a -D define that will update the USB descriptor sent to the host (there's no actual change to the app running, this is just a USB information change).

https://github.com/earlephilhower/arduino-pico/blob/fd1596b6c92d868dda0a7e47152298711f983b8c/cores/rp2040/RP2040USB.cpp#L270-L273

**Reply by maxgerhardt** [reply]
And for completeness' sake, the generating code for the USB max mA rating is in the makeboards.py

https://github.com/earlephilhower/arduino-pico/blob/354edb30d3014a96afa323d5fcd428a9a4992dac/tools/makeboards.py#L322-L327
If you were using Platformio, you would have to add
```ini
build_unflags = -DUSBD_MAX_POWER_MA=250
build_flags = -DUSBD_MAX_POWER_MA=500
```
to the `platformio.ini`.
But since you say
>I run this on VisualMicro under VisualStudio2019.

this extension will likely read the `boards.txt` of the Arduino core, so find the file locally and change
https://github.com/earlephilhower/arduino-pico/blob/354edb30d3014a96afa323d5fcd428a9a4992dac/boards.txt#L42
accordingly.

**Reply by stevecolt** [reply]
Well thank you both very much for your prompt replies.
Perhaps a little naughty but I just edited the boards.txt file as it was only changing 250 to 500 and that seemed to work according to USB TreeView so thanks a lot for your help.
Problem solved!!
Steve
